src/query_plan/operation.rs

```diff
@@ -31,7 +31,15 @@ pub enum NormalizedSelectionKey {
     Field {
         // field alias (if specified) or field name in the resulting selection set
         response_name: Name,
-        // directive applied on the field
+        // directives applied on the field
+        directives: DirectiveList,
+        // unique label/counter used to distinguish fields that cannot be merged
+        label: i32,
```

Comment:
is an i32 enough? not exactly sure how we use this, but do we think we need an atomic counter here? or will that be an overkill for its usecase?

Reply:
This is only unique within one document, right? I’d expect any document to reach the parser’s token limit well before it contains two billion selections.

Reply:
For now it's unique within the document. That will change once https://github.com/apollographql/federation-next/issues/103 is implemented (but I think the switch to `i64` can be put off till then).

Reply:
^ sgtm